change slice declarations to array initialization

mstrYoda opened the pull request:
Signed-off-by: Emre Savcı <emre.savci@trendyol.com>

Please ensure your pull request adheres to the following guidelines:

- [X] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [X] All code is covered by unit and/or runtime tests where feasible.
- [X] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [X] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [X] Thanks for contributing!

sayboras reviewed (approved):
Can you help to trim the commit description to something shorter ? Details can be added in commit message. Please refer to this [page](https://docs.cilium.io/en/latest/contributing/development/contributing_guide/#submitting-a-pull-request) for example.

The changes look good to me. Thanks.

Commit:
add array initialization

Change slice declarations to array initialization when len is already obvious.
This is a tiny performance improvement.

Signed-off-by: Emre Savcı <emre.savci@trendyol.com>
